### ISSUE TITLE
Visibility property returns incorrect information

### ISSUE THREAD

**Issue by leewillis77** [woocommerce/woocommerce#12370]
## EXPLANATION OF THE ISSUE

I'm testing the product CRUD stuff, and am seeing some inconsistent data with the visibility property on the master branch vs. 2.6.8 (and all previous versions for that matter). 

## STEPS TO REPRODUCE THE ISSUE

* Create a variable product
* Take a note of one of the variation IDs
* Save the script below as visibility.php, setting your variation ID where I've used 257
* Run the script with wp-cli against WooCommerce 2.6.8, then master

```
<?php

$variation_id = 257;
$factory = new WC_Product_Factory();
$variation_product = $factory->get_product( $variation_id );

var_dump($variation_product->visibility);
```

Running against WooCommerce 2.6.8, you get the following output, which is correct:
```
$ wp eval-file visibility.php
string(7) "visible"
```

Swapping WooCommerce to latest master, you get the following output, which is incorrect:
```
$ wp eval-file visibility.php
string(6) "hidden"
```

**Comment by claudiosanches** [comment]
Closed in favor of #12371
Thanks.

**Comment by leewillis77** [comment]
Thanks!
